### ISSUE TITLE
dns req flooding

### ISSUE THREAD

**Issue by cirocosta** [cirocosta/stress#4]

**Comment by galvez** [comment]
@cirocosta any plans for tackling this one?

**Comment by cirocosta** [comment]
Hey @galvez , not really, even though it's an interesting one.

At the time I was going to perform A record queries as I was curious about how the docker internal DNS server would react.

Maybe something in the area of:

```
create N threads which form the worker pool
on each worker:
  - use epoll up to a limit of inflight non-blocking queries
```
 
Are you looking for something else in particular or does this scenario fits your needs?

**Comment by galvez** [comment]
@cirocosta Nope, just something to add my 2018 study todo list. I liked the project and have been looking for things to exercise my C. Aaaand am fairly familiar with DNS. Thanks for the pointers, you've been doing an awesome work.

**Comment by cirocosta** [comment]
That's cool! Thanks!

By the time I coursed networking I worked though this book: http://www.unpbook.com/

It's very "C" based and unix oriented so I highly recommend.

It doesn't cover some "new" stuff that is mentioned in the c10k problem (http://www.kegel.com/c10k.html) but it's pretty useful as a foundation.

Have a good one!
